pkg/agent/agent.go

```diff
@@ -206,7 +165,54 @@ func (k *Klocksmith) Run() error {
 	k.lc.Reboot(false)
 
 	// cross fingers
-	time.Sleep(24 * 7 * time.Hour)
+	sleepOrDone(24*7*time.Hour, stop)
+	return nil
```

Comment:
Yeah, good point about the Daemonset being rescheduled. We probably would want to leave the sleepOrDone.